### PR TITLE
extended the "avatar" functionality

### PR DESCRIPTION
**Pull request by pedily**
- introduced a seperate avatar for messages from "agents"
- changed the logic that decides which avatar to use
- added the functionality to set user avatar via setting (`userAvatarUrl`)
- added the functionality to set the agent avatar via setting (`agentAvatarUrl`)
- added the functionality to set the agent avatar via override (`data._webchat.agentAvatarOverrideUrl`)
- changed the avatar for the typing status indicator to not show any content (as we do not know whether a message will come from an agent or the bot)

Generally, this commit streamlined the avatar logic for user, bot and agent messages. They all work the same now. They follow the schema "override > setting > default" for choosing avatars for messages.

### CODE REVIEW

File: docs/embedding.md [hunk 1]
```diff
@@ -58,9 +58,12 @@ See it in action:
 | enableSTT | boolean | false | Whether to enable speech input that lets the user speak to the Webchat instead of only typing. 
 | enableTTS | boolean | false | Whether to enable the browser to read the bot messages aloud. 
 | colorScheme | string | The background color of the header and bot messages in the Webchat. | designTemplate | 1 or 2 | 1 | The Webchat design template to use. We default to design template 1 (bottom right with a button), you can switch to template 2, which is the centered webchat. 
-| messageLogoUrl | string | COGNIGY.AI Logo | The logo to display next to the messages from your bot in the Webchat. Defaults to a COGNIGY.AI logo. 
+| messageLogoUrl | string | COGNIGY.AI Logo | A custom avatar that should be displayed next to bot messages. Defaults to a COGNIGY.AI logo. 
+| userAvatarUrl | string | undefined | A custom avatar that should be displayed next to user messages. Defaults to a user icon.
+| agentAvatarUrl | astring | undefined | A custom avatar that sould be displayed next to agent messages
```
Comment (monty0157) [inline docs/embedding.md]: string*

File: src/webchat/store/messages/message-middleware.ts [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
-import { defaultBotAvatar, defaultUserImg } from "../../../webchat-ui";
+import { sanitizeHTML } from "../../helper/sanitize";
 import { SocketClient } from "@cognigy/socket-client";
 
+
+const defaultAgentAvatar = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGAAAABgCAQAAABIkb+zAAACOklEQVR4Ae3ZA2ydURwF8P9s2+bjPSdGo0aN08V+URtbL+a8BbO9xfZs2zaCuW7vbDx8uLfp/3dinw+XopRSSimllFJhYm9TjV08wwdoYB0f8ix2mDkTe0p7YIZxDeto/5I6rjHDxGtdkcc72n8H75CXruKn1CAcpi0cHE4NEv9kp+EubXHB3ew08QuH4hFt8cGj5Ajxx9hePE1bYi6k+4gvMJ+29GCe+CEzhvW0ZaQ+PVZ8wDW0ZWatuJfozrqyC9Qluotr2Sra8pOtEtewMkgBrBLXsC9QgX3iGm4EKnBDXOP7QAXeiWt4G6jAW3ENNwMVuCmu4UCgAgc6/DCqE1miO9+7X0oEgtVlF1gjPkiOKHs5Pbx9b2jme7SlxPmSC5we20v8kRjJh6Vt6jlU/JKZztsBj1XcH2zxGG3h4ERqkPgp0R35AhvMOuQT3cVnyRH/O9wt4zjLzaj00/F6/dfj9WrPj9eVUkqpRPeMMTnMxxbu4fWf5uP3uME93IZ5JpcxHi4lzGjWYgPPsom2cNDIs9jAWjNaXJvaw1RyES/SlpmLXGQqHb0Rgsv5hjaEvOJyIt6lWg4nacMNTppcHMu9LqYGL2ijCZ6bGuki0TEVuEIbbXDFVEgU2JsbaWPKRvYOf6C8SBtjLoY6yKbH4h5tvMHd5DgJR6Ivb9E6yK1EX6c3AMGDlRIcZtG6i5ktQWGpywJYKkHxgtMC5yUo1tM6TL0ERes2WkALaAEtEEm0gFJKKaWUUkp9ABvn3SEbw3cFAAAAAElFTkSuQmCC";
+const defaultBotAvatar = "https://s3.eu-central-1.amazonaws.com/cognigydev/CognigyWebchat/images/cognigy_logo.svg"
```
Comment (monty0157) [inline src/webchat/store/messages/message-middleware.ts]: Why are we loading this image from S3?

File: src/webchat/store/messages/message-middleware.ts [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
+const defaultUserAvatar = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGAAAABgCAQAAABIkb+zAAACOklEQVR4Ae3ZA2ydURwF8P9s2+bjPSdGo0aN08V+URtbL+a8BbO9xfZs2zaCuW7vbDx8uLfp/3dinw+XopRSSimllFJhYm9TjV08wwdoYB0f8ix2mDkTe0p7YIZxDeto/5I6rjHDxGtdkcc72n8H75CXruKn1CAcpi0cHE4NEv9kp+EubXHB3ew08QuH4hFt8cGj5Ajxx9hePE1bYi6k+4gvMJ+29GCe+CEzhvW0ZaQ+PVZ8wDW0ZWatuJfozrqyC9Qluotr2Sra8pOtEtewMkgBrBLXsC9QgX3iGm4EKnBDXOP7QAXeiWt4G6jAW3ENNwMVuCmu4UCgAgc6/DCqE1miO9+7X0oEgtVlF1gjPkiOKHs5Pbx9b2jme7SlxPmSC5we20v8kRjJh6Vt6jlU/JKZztsBj1XcH2zxGG3h4ERqkPgp0R35AhvMOuQT3cVnyRH/O9wt4zjLzaj00/F6/dfj9WrPj9eVUkqpRPeMMTnMxxbu4fWf5uP3uME93IZ5JpcxHi4lzGjWYgPPsom2cNDIs9jAWjNaXJvaw1RyES/SlpmLXGQqHb0Rgsv5hjaEvOJyIt6lWg4nacMNTppcHMu9LqYGL2ijCZ6bGuki0TEVuEIbbXDFVEgU2JsbaWPKRvYOf6C8SBtjLoY6yKbH4h5tvMHd5DgJR6Ivb9E6yK1EX6c3AMGDlRIcZtG6i5ktQWGpywJYKkHxgtMC5yUo1tM6TL0ERes2WkALaAEtEEm0gFJKKaWUUkp9ABvn3SEbw3cFAAAAAElFTkSuQmCC"
+
 export interface ISendMessageOptions {
-    /* overrides the displayed text within a chat bubble. useful for e.g. buttons */
+    /* overrides the displayed text within a chat bubble. useful for e.g. b
+import { IWebchatConfig } from "../../../common/interfaces/webchat-config";uttons */
```
Comment (monty0157) [inline src/webchat/store/messages/message-middleware.ts]: ? :D

File: src/webchat/store/messages/message-middleware.ts (hunk 2 above)
Comment (pedily) [reply src/webchat/store/messages/message-middleware.ts]: not sure. i just moved these lines in this PR. we can also include the logo into the source

File: src/webchat/store/messages/message-middleware.ts (hunk 3 above)
Comment (pedily) [reply src/webchat/store/messages/message-middleware.ts]: wat. let me check that. damn merges.

File: src/webchat/store/messages/message-middleware.ts (hunk 2 above)
Comment (monty0157) [reply src/webchat/store/messages/message-middleware.ts]: We should. This is technically out of scope for this PR, but we should definitely do this soon

Comment (pedily) [reply src/webchat/store/messages/message-middleware.ts]: i will convert it into a data url and put it here

File: docs/embedding.md (hunk 1 above)
Comment (pedily) [reply docs/embedding.md]: fixed

File: src/webchat/store/messages/message-middleware.ts (hunk 3 above)
Comment (pedily) [reply src/webchat/store/messages/message-middleware.ts]: fixed

File: src/webchat/store/messages/message-middleware.ts (hunk 2 above)
Comment (pedily) [reply src/webchat/store/messages/message-middleware.ts]: fixed

**Comment by Cesarcs7** [comment]
lgtm

**Pull request merged**